### PR TITLE
Code Suggestions

### PR DESCRIPTION
**Pull request by Sparky200**
See Commit #28.

Most of these were done automatically by IJ, and some by me.

### CODE REVIEW

File: src/main/java/io/github/railroad/config/LanguageConfig.java [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
 
 public class LanguageConfig extends AbstractConfig {
 
 	private Map<String, String> languageTranslator;
 
 	public LanguageConfig(String file) {
-		languageTranslator = new HashMap<String, String>();
-		JSONTokener tokener = new JSONTokener(Railroad.class.getResourceAsStream(file));
-		JSONObject obj = new JSONObject(tokener);
+		languageTranslator = new HashMap<>();
```
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/config/LanguageConfig.java:15]: why did you remove the `String, String`?

File: src/main/java/io/github/railroad/config/SyntaxConfig.java [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import java.io.File;
+import java.util.*;
 
 public class SyntaxConfig extends AbstractConfig {
 
-	public List<SyntaxObject> languages = new ArrayList<SyntaxObject>();
+	public List<SyntaxObject> languages = new ArrayList<>();
```
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/config/SyntaxConfig.java:16]: same here. why are you leaving it like that. just remove the <> completely?

File: src/main/java/io/github/railroad/config/LanguageConfig.java (hunk 1 above)
Comment (Sparky200) [reply src/main/java/io/github/railroad/config/LanguageConfig.java:15]: It's repetitive

File: src/main/java/io/github/railroad/config/SyntaxConfig.java (hunk 2 above)
Comment (Sparky200) [reply src/main/java/io/github/railroad/config/SyntaxConfig.java:16]: <> is required, but inserting the value is redundant. You can take IJ's word for it, too. I do this too, IJ just picks up on it.

Comment (DaRealTurtyWurty) [reply src/main/java/io/github/railroad/config/SyntaxConfig.java:16]: its most certainly not required. you either leave it raw or keep the generics all there, leaving it blank makes no sense at all

Comment (Sparky200) [reply src/main/java/io/github/railroad/config/SyntaxConfig.java:16]: I mean, yes, but there's also no reason to *keep* it. Don't take it from me; IJ's suggestions are submitted by experts, look it up.

File: src/main/java/io/github/railroad/debugger/syntax/SyntaxHandler.java [hunk 3]
```diff
@@ -31,6 +30,7 @@ public static void main(String[] args) {
 	private ExecutorService executor;
 
 	@SuppressWarnings("unused")
+	// @SuppressWarnings Should be removed, usually if you have to use SuppressWarnings, you have something to fix :)
```
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/debugger/syntax/SyntaxHandler.java:33]: I feel like this is obvious? If it's unused then we know we have something to fix. We are just removing a stupid warning from our project.

File: src/main/java/io/github/railroad/config/SyntaxConfig.java [hunk 4]
```diff
@@ -48,7 +45,7 @@ public SyntaxConfig() {
 				JSONObject obj = new JSONObject(tokenizer);
 				JSONArray rules = obj.getJSONArray("rules");
 
-				Map<String, EnumSyntaxType> ruleMap = new HashMap<>();
+				Map<String, EnumSyntaxType> ruleMap = new HashMap();
```
Comment (Sparky200) [inline src/main/java/io/github/railroad/config/SyntaxConfig.java:48]: IJ's gonna yell at me if you make it raw >:(
IJ scares me man 🤣

File: src/main/java/io/github/railroad/debugger/syntax/SyntaxHandler.java (hunk 3 above)
Comment (Sparky200) [reply src/main/java/io/github/railroad/debugger/syntax/SyntaxHandler.java:33]: I'm not trying to say it's bad or anything, just wanted to point it out 😛

File: src/main/java/io/github/railroad/config/SyntaxConfig.java (hunk 4 above)
Comment (DaRealTurtyWurty) [reply src/main/java/io/github/railroad/config/SyntaxConfig.java:48]: i mean. just turn off the raw types warning lol

Comment (Sparky200) [reply src/main/java/io/github/railroad/config/SyntaxConfig.java:48]: I could... but the warning is there for a reason :wink:

Comment (DaRealTurtyWurty) [reply src/main/java/io/github/railroad/config/SyntaxConfig.java:48]: and you are just trying to bypass that reason by doing `<>` xd

Comment (Sparky200) [reply src/main/java/io/github/railroad/config/SyntaxConfig.java:48]: It's not bypassing. Including it tells java to look at the declaration.

File: src/main/java/io/github/railroad/utility/FileUtils.java [hunk 5]
```diff
@@ -1,48 +1,38 @@
...
 
-	public static File createFolder(String pathIn) {
+	public static File createFolder(String pathIn) { // Unused Method
 		File newFolder = new File(pathIn);
-		newFolder.mkdir();
+		newFolder.mkdir(); // Return is ignored. Add check.
 		return newFolder;
 	}
 
-	public static void deleteFile(File file) {
```
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/utility/FileUtils.java:25]: Why did you remove this?

Comment (Sparky200) [reply src/main/java/io/github/railroad/utility/FileUtils.java:25]: The method doesn't *do* anything. It's not even a shorthand. You can just call what it calls, since you pass in `file` anyway.

Comment (DaRealTurtyWurty) [reply src/main/java/io/github/railroad/utility/FileUtils.java:25]: i mean thats fair enough. but in the future, this will be added back. because this method wont just do this, this was just a template. it will also need to update things

**Review by DaRealTurtyWurty — APPROVED** [review]

File: src/main/java/io/github/railroad/utility/FileUtils.java (hunk 5 above)
Comment (Sparky200) [reply src/main/java/io/github/railroad/utility/FileUtils.java:25]: Sounds good. Probably should include a JavaDoc that says that :wink:
